src/components/kubectl/autoversion.ts

```diff
@@ -101,11 +101,12 @@ async function ensureCacheIsForCurrentKubeconfig(): Promise<void> {
 }
 
 async function getKubeconfigPathHash(): Promise<string | undefined> {
-    const kubeconfigPath = getActiveKubeconfig() || process.env['KUBECONFIG'] || path.join(shell.home(), '.kube/config');
-    if (!await fs.existsAsync(kubeconfigPath)) {
+    const kubeconfigPath = getKubeconfigPath();
+    const kubeconfigFilePath = kubeconfigPath.pathType === "host" ? kubeconfigPath.hostPath : kubeconfigPath.wslPath;
+    if (!await fs.existsAsync(kubeconfigFilePath)) {
```

Comment:
This has surfaced an issue in this procedure - in WSL mode, the path will be a Unix path, but `fs.existsAsync` will still look in the host filesystem.  I don't think you need to fix this, as the bug was already there but latent, and the impact is to defeat caching rather than to do anything calamitous, but I wanted to note it for future work.

Reply:
Added a TODO!